pkg/ebpf/c/tracee.bpf.c

```diff
@@ -555,32 +557,54 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
-        int parent_ns_pid = get_task_ns_pid(parent);
-        int parent_ns_tgid = get_task_ns_tgid(parent);
-        int child_ns_pid = get_task_ns_pid(child);
-        int child_ns_tgid = get_task_ns_tgid(child);
-
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_pid, sizeof(int), 0);
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_pid, sizeof(int), 1);
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_tgid, sizeof(int), 2);
-        save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_tgid, sizeof(int), 3);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_pid, sizeof(int), 4);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_pid, sizeof(int), 5);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_tgid, sizeof(int), 6);
-        save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_tgid, sizeof(int), 7);
-        save_to_submit_buf(&p.event->args_buf, (void *) &start_time, sizeof(u64), 8);
+    int parent_ns_pid = get_task_ns_pid(parent);
+    int parent_ns_tgid = get_task_ns_tgid(parent);
+    int child_ns_pid = get_task_ns_pid(child);
+    int child_ns_tgid = get_task_ns_tgid(child);
+
+    // parent
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_pid, sizeof(int), 0);
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_pid, sizeof(int), 1);
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_tgid, sizeof(int), 2);
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_ns_tgid, sizeof(int), 3);
+    // child
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_pid, sizeof(int), 4);
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_pid, sizeof(int), 5);
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_tgid, sizeof(int), 6);
+    save_to_submit_buf(&p.event->args_buf, (void *) &child_ns_tgid, sizeof(int), 7);
+    // start times
+    save_to_submit_buf(&p.event->args_buf, (void *) &task_start_time, sizeof(u64), 8);
+
+    // Submit the regular SCHED_PROCESS_FORK event
 
+    if (should_submit(SCHED_PROCESS_FORK, p.event) || p.config->options & OPT_PROCESS_INFO) {
         events_perf_submit(&p, SCHED_PROCESS_FORK, 0);
     }
 
-    return 0;
+    //
+    // Submit SCHED_PROCESS_FORK signal event now (only args, other perfbuffer)
+    //
+
+    // args: parent_start_time used by control plane ONLY
+    save_to_submit_buf(&p.event->args_buf, (void *) &parent_start_time, sizeof(u64), 9);
+
+    controlplane_signal_t *signal = init_controlplane_signal();
+    if (unlikely(signal == NULL))
+        return 0;
+
+    // ATTENTION:
+    // 1. __builtin_memcpy won't work in between 2 eBPF maps
+    // 2. calling save_to_submit_buf(signal) would raise instr count too much
+    buffer_memcpy(&signal->args_buf, &p.event->args_buf, sizeof(signal->args_buf));
+
+    return signal_perf_submit(ctx, signal, SCHED_PROCESS_FORK);
```

Comment:
I remember reading some time ago that the return value of an eBPF program has significance in non-packet programs too, related to its cleanup or something along those lines. That is why I set the return values of the control plane to always 0 (I see you've done that change there as well).
So I just want to make sure this is fine.

Reply:
Or we can still return 0 to make sure we don't brake existing behavior. From what I know LSM hooks return code define execution flow changes (as well as network related programs). There might be other cases but tracing shouldn't change iirc. Either way will return 0.